package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1266,13 +1266,13 @@ def chi1_selections(residues, n_name='N', ca_name='CA', cb_name='CB',
         """
         results = np.array([None]*len(residues))
         names = [n_name, ca_name, cb_name, cg_name]
-        keep = [all(sum(r.atoms.names == n) == 1 for n in names)
-                for r in residues]
+        keep = [all(sum(np.in1d(r.atoms.names, n.split())) == 1
```

Comment:
I like your updated code.

@IAlibay @richardjgowers are we ok with also updating `chi1_selection()` with the same solution, which avoids `select_atoms()`? We could do a separate PR but this seems unnecessary bureaucracy to me, unless we want a cleaner separation of work in CHANGELOG.

Reply:
I was thinking about the CHANGELOG; I had a couple ideas:

1. List the changes to both `chi1_selections` and `chi1_selection` in a single entry in the Fixes section. Maybe something like this: "_Fix chi1_selections incorrectly returning None (Issue 4108) and chi_selection running unnecessarily slowly_".
2. List the change to `chi1_selections` as a Fix, and the change to `chi1_selection` as an enhancement.   Maybe "_Fix chi1_selections incorrectly returning None (Issue 4108)_" in Fixes, and "_Improved speed of chi1_selection_" in Enhancements.

I think I favor option (2).  But I don't know the history or culture behind this project's CHANGELOG, so I don't have a strong opinion.  And it's definitely possible there's a better option that I haven't thought of!

Reply:
(2) seems better to me

Reply:
@DanielJamesEvans I haven't heard any opposition to you updating `chi1_selection()` in this PR. I encourage you to add it to your PR. Once you've done so, please ping me so that I can review again.

Obviously, anyone else is also welcome to review and if dissenting opinions are voiced in other reviews we can address them then.

Reply:
Sounds good!  I added the `chi1_selection()` update to my PR.